src/driver.rs

```diff
@@ -210,6 +212,23 @@ pub enum OutputFormat {
     Format,
 }
 
+impl FromStr for OutputFormat {
+    type Err = &'static str;
+
+    fn from_str(s: &str) -> StdResult<Self, Self::Err> {
+        let actual = match s {
+            "aux" => OutputFormat::Aux,
+            "html" => OutputFormat::Html,
+            "xdv" => OutputFormat::Xdv,
+            "pdf" => OutputFormat::Pdf,
+            "fmt" => OutputFormat::Format,
+            _ => unreachable!(),
```

Comment:
I would have expected these converters to return an `Err` with an unexpected string, rather than panic. Is there a specific reason for the use of `unreachable!`? If so, explanatory comments would be helpful.

Reply:
> Is there a specific reason for the use of unreachable!? If so, explanatory comments would be helpful.

I wondered about this as well. I actually tried to maintain the existing [structure](https://github.com/tectonic-typesetting/tectonic/blob/master/src/bin/tectonic.rs#L42) but given the semantics has changed I should update this error to be more meaningful.

Reply:
Yeah ... in that targeted usage, we know that clap/structopt will prevent the code from ever being sent an unacceptable value. For a broader `FromString` implementation, though, we can't make that assumption.